File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/module/project/view/parameters/BugScreenData.java
package com.esofthead.mycollab.mobile.module.project.view.parameters;

import com.esofthead.mycollab.module.tracker.domain.BugWithBLOBs;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.5.2
 *
 */
public class BugScreenData {

	public static class Search extends ScreenData<BugFilterParameter> {

		public Search(BugFilterParameter params) {
			super(params);
		}
	}

	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<BugWithBLOBs> {

		public Add(BugWithBLOBs params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<BugWithBLOBs> {

		public Edit(BugWithBLOBs params) {
			super(params);
		}
	}
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/module/project/view/parameters/BugScreenData.scala
package com.esofthead.mycollab.mobile.module.project.view.parameters

import com.esofthead.mycollab.module.tracker.domain.BugWithBLOBs
import com.esofthead.mycollab.module.tracker.domain.criteria.BugSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object BugScreenData {

    class Search(param: BugSearchCriteria) extends ScreenData(param) {}

    class Read(param: Integer) extends ScreenData(param) {}

    class Add(param: BugWithBLOBs) extends ScreenData(param) {}

    class Edit(param: BugWithBLOBs) extends ScreenData(param) {}

}
